File: src/test/java/net/flyingfishflash/loremlist/unit/core/response/structure/DispositionTests.java
package net.flyingfishflash.loremlist.unit.core.response.structure;

import static org.assertj.core.api.Assertions.assertThat;

import net.flyingfishflash.loremlist.core.response.structure.DispositionOfProblem;
import org.junit.jupiter.api.Test;
import org.springframework.http.HttpStatus;

class DispositionTests {

  @Test
  void failureWhenHttpStatusIs4xx() {
    assertThat(DispositionOfProblem.calcDisposition(HttpStatus.I_AM_A_TEAPOT))
        .isEqualTo(DispositionOfProblem.FAILURE);
  }

  @Test
  void errorWhenHttpStatusIs5xx() {
    assertThat(DispositionOfProblem.calcDisposition(HttpStatus.INTERNAL_SERVER_ERROR))
        .isEqualTo(DispositionOfProblem.ERROR);
  }

  @Test
  void undefinedWhenHttpStatusIsNot4xxOr5xx() {
    assertThat(DispositionOfProblem.calcDisposition(HttpStatus.CONTINUE))
        .isEqualTo(DispositionOfProblem.UNDEFINED);
    assertThat(DispositionOfProblem.calcDisposition(HttpStatus.OK))
        .isEqualTo(DispositionOfProblem.UNDEFINED);
    assertThat(DispositionOfProblem.calcDisposition(HttpStatus.MULTIPLE_CHOICES))
        .isEqualTo(DispositionOfProblem.UNDEFINED);
  }
}

File: src/test/kotlin/net/flyingfishflash/loremlist/unit/core/response/structure/DispositionTests.kt
package net.flyingfishflash.loremlist.unit.core.response.structure

import io.kotest.core.spec.style.DescribeSpec
import io.kotest.matchers.shouldBe
import net.flyingfishflash.loremlist.core.response.structure.DispositionOfProblem
import org.springframework.http.HttpStatus

class DispositionTests : DescribeSpec({
  describe("calcDisposition()") {
    it("failure when http status is 4xx") {
      DispositionOfProblem.calcDisposition(HttpStatus.I_AM_A_TEAPOT)
        .shouldBe(DispositionOfProblem.FAILURE)
    }

    it("error when http status is 5xx") {
      DispositionOfProblem.calcDisposition(HttpStatus.INTERNAL_SERVER_ERROR)
        .shouldBe(DispositionOfProblem.ERROR)
    }

    it("undefined when http status is not 4xx or 5xx") {
      // 100
      DispositionOfProblem.calcDisposition(HttpStatus.CONTINUE)
        .shouldBe(DispositionOfProblem.UNDEFINED)
      // 200
      DispositionOfProblem.calcDisposition(HttpStatus.OK)
        .shouldBe(DispositionOfProblem.UNDEFINED)
      // 300
      DispositionOfProblem.calcDisposition(HttpStatus.MULTIPLE_CHOICES)
        .shouldBe(DispositionOfProblem.UNDEFINED)
    }
  }
})
